File: src/main/java/com/mini/credit/dto/caisse/OperationCaisseResponse.java
package com.mini.credit.dto.caisse;

import com.mini.credit.enums.CategorieOperationCaisse;
import com.mini.credit.enums.ModePaiement;
import com.mini.credit.enums.TypeOperationCaisse;
import lombok.Builder;
import lombok.Getter;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Getter
@Builder
public class OperationCaisseResponse {
    private Long id;
    private String numeroPiece;
    private Long sessionCaisseId;
    private Long caisseId;
    private LocalDateTime dateOperation;
    private TypeOperationCaisse typeOperation;
    private CategorieOperationCaisse categorieOperation;
    private BigDecimal montant;
    private String devise;
    private Long membreId;
    private Long creditId;
    private Long remboursementId;
    private Long operationEpargneId;
    private Long paiementCreditId; // ✅ AJOUT
    private Long agentId;
    private Long createdById; // ✅ AJOUT
    private ModePaiement modePaiement; // ✅ AJOUT
    private String description;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: frontend/src/app/features/caisse/models/operation-caisse-response.ts
import { ModePaiement } from '../../epargne/models/mode-paiement';
import { CategorieOperationCaisse } from './categorie-operation-caisse';
import { NatureFinancementApprovisionnement } from './nature-financement-approvisionnement';
import { SourceOperationCaisse } from './source-operation-caisse';
import { TypeOperationCaisse } from './type-operation-caisse';

export interface OperationCaisseResponse {
  id: number;
  numeroPiece: string;
  sessionCaisseId: number;
  caisseId: number;
  dateOperation: string;

  typeOperation: TypeOperationCaisse;
  categorieOperation: CategorieOperationCaisse;
  natureFinancement?: NatureFinancementApprovisionnement;

  /** PATCH 7 — Source de tracabilité retournée par le backend. */
  source?: SourceOperationCaisse;

  /** PATCH 7 — Référence externe lisible selon la source. */
  referenceExterne?: string;

  montant: number;
  devise: string;

  membreId?: number;
  creditId?: number;
  remboursementId?: number;
  operationEpargneId?: number;
  paiementCreditId?: number;

  agentId?: number;
  createdById?: number;
  modePaiement?: ModePaiement;
  description?: string;

  createdAt: string;
  updatedAt: string;
}